rust/lance-linalg/src/matrix.rs

```diff
@@ -52,12 +55,25 @@ where
     pub transpose: bool,
 }
 
-impl<T: FloatArray> MatrixView<T>
+impl<T: ArrowFloatType> Clone for MatrixView<T>
 where
-    Standard: rand::distributions::Distribution<<T as FloatArray>::Native>,
+    Standard: rand::distributions::Distribution<<T as ArrowFloatType>::Native>,
```

Comment:
Do you need the rand constraint for clone?

Reply:
Matrix has a random initialization method that needs this constraint. 

https://github.com/lancedb/lance/blob/27814c56fd99241ce4ded977896179746d27c74d/rust/lance-linalg/src/matrix.rs#L69